src/content/content_manager.cc

```diff
@@ -1225,6 +1225,9 @@ std::shared_ptr<CdsObject> ContentManager::createObjectFromFile(const fs::direct
         obj->setTitle(f2i->convert(title));
 
         MetadataHandler::extractMetaData(context, item, dirEnt);
+        if (item->getMetaData(M_DATE).empty())
+            item->addMetaData(M_DATE, to_dcDate(dirEnt.last_write_time(ec)));
+
```

Comment:
the `last_write_time` is in mtime already, so you can just
`fmt::format("{:%FT%T%z}", fmt::localtime(item->getMTime().count()))`